acai-ios/BowlOrderDetailViewController.swift

```diff
@@ -0,0 +1,116 @@
+        
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.backgroundColor = .white
+        collectionView.showsVerticalScrollIndicator = false
+        view.addSubview(collectionView)
+        collectionView.snp.makeConstraints { (make) in
+            make.edges.equalToSuperview()
+        }
+        
+        collectionViewListAdapter = ListAdapter(updater: ListAdapterUpdater(), viewController: self, workingRangeSize: 0)
+        collectionViewListAdapter.collectionView = collectionView
+        collectionViewListAdapter.dataSource = self
+    }
+}
+
+extension BowlOrderDetailViewController: ListAdapterDataSource {
+    func objects(for listAdapter: ListAdapter) -> [ListDiffable] {
+    
+        //hard code
+        
+        bowlItem = Acai.testBowl
+        
+        toppingOptions = bowlItem.toppingOptions
+        baseOptions = bowlItem.baseOptions
+        var list: [ListDiffable] = []
+        list.append("Choose your base" as ListDiffable)
+        list.append(Acai.bowlOrderBaseOptionsIdentifier as ListDiffable)
+        list.append("Choose your toppings" as ListDiffable)
+        list.append(Acai.bowlOrderToppingOptionsIdentifier as ListDiffable)
+        return list
+    }
+    
+    func listAdapter(_ listAdapter: ListAdapter, sectionControllerFor object: Any) -> ListSectionController {
+        let orderCustomizationListSectionController = OrderCustomizationListSectionController()
+        if let object = object as? String, object == Acai.bowlOrderBaseOptionsIdentifier {
+            orderCustomizationListSectionController.options = bowlItem.baseOptions
+            orderCustomizationListSectionController.deselectOptionDelegate = self
+            orderCustomizationListSectionController.selectOptionDelegate = self
+        }
+        else if let object = object as? String, object == Acai.bowlOrderToppingOptionsIdentifier {
+            orderCustomizationListSectionController.options = bowlItem.toppingOptions
+            orderCustomizationListSectionController.selectOptionDelegate = self
+        }
+        else if let object = object as? String {
+            let headerListSectionController = HeaderListSectionController()
+            headerListSectionController.title = object
+            return headerListSectionController
+        }
+        return orderCustomizationListSectionController
+    }
+    
+    func emptyView(for listAdapter: ListAdapter) -> UIView? {
+        return nil
+    }
+}
+
+protocol DidDeselectOption: class {
+    func deselectOption(at index: Int)
+}
+
+protocol DidSelectOption: class {
+    func selectOption(at index: Int, type: CustomizationOptionType)
+}
+
+extension BowlOrderDetailViewController: DidDeselectOption, DidSelectOption {
+    
+    func deselectOption(at index: Int) {
+        baseOptions[index].isSelected = false
+        collectionView.reloadData()
```

Comment:
one of the biggest reasons we use `IGListKit` is so that we don't have to call `reloadData`. `reloadData` is expensive because it recreates every single cell. you should be updating your `ListDiffable` models and then calling `listAdapter.performUpdates()`